### ISSUE TITLE
Android <= 6 crashes because of notifications

### ISSUE THREAD

**Issue by almasaeed2010** [statonlab/Treesnap-mobile#520]
During the butterfly festival today, one device crashed and this is the error we got:

```
Fatal Exception: java.lang.SecurityException: !@Too many alarms (500) registered from pid 10252 uid 10053
       at android.os.Parcel.readException(Parcel.java:1620)
       at android.os.Parcel.readException(Parcel.java:1573)
       at android.app.IAlarmManager$Stub$Proxy.set(IAlarmManager.java:217)
       at android.app.AlarmManager.setImpl(AlarmManager.java:484)
       at android.app.AlarmManager.setExact(AlarmManager.java:422)
       at com.dieam.reactnativepushnotification.modules.RNPushNotificationHelper.sendNotificationScheduledCore(RNPushNotificationHelper.java:127)
       at com.dieam.reactnativepushnotification.modules.RNPushNotificationHelper.sendNotificationScheduled(RNPushNotificationHelper.java:114)
       at com.dieam.reactnativepushnotification.modules.RNPushNotification.scheduleLocalNotification(RNPushNotification.java:135)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:160)
       at com.facebook.react.bridge.queue.NativeRunnable.run(NativeRunnable.java)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29)
       at android.os.Looper.loop(Looper.java:158)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:192)
       at java.lang.Thread.run(Thread.java:818)
```

**Comment by almasaeed2010** [comment]
So this issue seems to happen only to Samsung devices according to many reports of the issue online.

I am not entirely sure how to handle this yet.

**Comment by almasaeed2010** [comment]
Fixed in #521 

Since we don't have a samsung device with API <= 23, we can't test this fix. We'll have to publish and see if crash reports continue to come in.
